src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,45 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($day-width * 7) + ($gap * 6) + ($padding * 2);
+  height: 100%;
+  padding: $padding;
+  gap: $gap;
+  justify-content: flex-start;
+  align-items: center;
+  margin: 0 auto;
+
+  @each $day, $num in $week {
+    &--start-#{$day} > :first-child {
+      margin-left: ($day-width + $gap) * $num;
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &--month-#{$index} > :nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    width: $day-width;
+    height: $day-width;
+    background-color: #eee;
+    transition: 0.5s;
```

Comment:
It is better to specify for which properties you want to apply the transition. 
Now it applies to all the properties, which is not good for performance
https://stackoverflow.com/questions/8947441/css3-transitions-is-transition-all-slower-than-transition-x